src/utils/quote_report_utils.ts

```diff
@@ -83,6 +84,7 @@ export function publishQuoteReport(
                 logOpts.submissionBy === 'taker' || logOpts.submissionBy === 'gaslessSwapAmm'
                     ? logOpts.decodedUniqueId
                     : undefined,
+            ammQuoteUniqueId: logOpts.submissionBy === 'gaslessSwapAmm' ? logOpts.ammQuoteUniqueId : undefined,
```

Comment:
this term `gaslessSwapAmm` is becoming a bit of a magic value - can we pull it into a constant? Do you also understand what `taker` vs `gaslessSwapAmm` is - and if it makes sense to group them, an enum might be better to represent them

Reply:
The current way `submissionBy` field is defined: `submissionBy: 'taker' | 'gaslessSwapAmm' | 'rfqm'`

`gaslessSwapAmm` means this quote report is for gasless swap AMM quote. `taker` means `/swap`. For now, I personally think either way (the current way or enum) works fine. If we were to make changes on this, thinking about doing it in a separate PR as otherwise this PR would contain different set of changes